doc/source/whatsnew/v1.2.0.rst

```diff
@@ -8,6 +8,20 @@ including other versions of pandas.
 
 {{ header }}
 
+.. warning::
+
+   Previously, the default argument ``engine=None`` to ``pd.read_excel``
+   would result in using the xlrd engine in many cases. The engine xlrd is no longer
+   maintained, and is not supported with python >= 3.9. When ``engine=None``, the
+   following logic is now used to determine the engine.
+
+   - If ``path_or_buffer`` is an OpenDocument format (.odf, .ods, .odt), then odf will be used.
+   - Otherwise if ``path_or_buffer`` is a bytes stream, the file has the extension ``.xls``, or is an xlrd Book instance, then xlrd will be used.
+   - Otherwise if openpyxl is installed, then openpyxl will be used.
+   - Otherwise xlrd will be used and a ``FutureWarning`` will be raised.
```

Comment:
I would maybe rearrange this list: the most important piece of information we want to convey here is that for xlsx files the default changed from xlrd to openpyxl, if installed. So I would also put that on top of the list (or keep it just to this for the whatsnew, as the other items didn't change. The full list is still in the actual docs).

Reply:
We don't actually look at the extension or the file format when determining the engine in various cases, so it isn't just changing for xlsx files, right? What do you think of this:

```
Previously, the default argument ``engine=None`` to ``pd.read_excel``
would result in using the `xlrd <https://xlrd.readthedocs.io/en/latest/>`_ engine
in many cases. The engine ``xlrd`` is no longer
maintained, and is not supported with python >= 3.9. If 
`openpyxl <https://pypi.org/project/openpyxl/>`_  is installed, many of these 
cases will now default to using the ``openpyxl`` engine. See the
:func:`read_excel` docs for more details.
```